run.sh

```diff
@@ -0,0 +1,171 @@
+        echo "Error: $*"
+    fi
+    cat << EOF
+
+Usage: $PROGNAME [OPTION ...] [foo] [bar]
+
+-----------------------------------------------------------------
+|  CLI tool for building and managing Edgebox updates           |
+-----------------------------------------------------------------
+
+Options:
+-h, --help               display this usage message and exit
+-c, --check              check for updates
+-u, --update             update the system
+
+EOF
+    exit 1
+}
+
+# Update check. This function is called when the --check flag is passed
+# It should check for updates by doing the following:
+# 1. Open the file versions.env and read the current version for each component (ws, api, apps, logger, edgeboxctl)
+# 2. Check in each component folder for the available git tags
+# 3. If there is a newer (semver) tag than the one read from the file, save it to a variable
+# 4. Write the new available next version to the versions.env file
+check() {
+    echo "Checking for updates..."
+
+    # Remove the file targets.env if it exists
+    if [ -f $COMPONENTS_DIR/updater/targets.env ]; then
+        rm $COMPONENTS_DIR/updater/targets.env
+    fi
+
+    # Check for updates for each component
+    for component in ws api apps logger edgeboxctl; do
+        
+        # Convert to uppercase
+        component_upper=$(echo "$component" | tr '[:lower:]' '[:upper:]')
+
+        # echo "Checking for updates for $component_upper..."
+        cd $COMPONENTS_DIR/$component
+
+        # Check if versions.env file exists
+        if [ ! -f $COMPONENTS_DIR/updater/versions.env ]; then
+            # Try to get the current version from the git tags
+            current_version=$(git describe --tags --exact-match 2>/dev/null || true)
+
+        else
+            current_version=$(grep -E "^${component_upper}_VERSION=" /home/system/components/updater/versions.env | cut -d '=' -f2)
+        fi        
+
+        # echo "Current version: $current_version"
+        
+        # echo "Available versions:"
+        # git tag -l | sort -V
+
+        # Get the latest version
+        latest_version=$(git tag -l | sort -V | tail -1)
+        # echo "Latest version: $latest_version"
+
+        # Get the next version
+        next_version=$(git tag -l | sort -V | grep -A1 "$current_version" | tail -1)
+
+        # echo "Next version: $next_version"
+
+        # If there is a current version, do further checks
+        if [ "$current_version" != "" ]; then
+            # If the current version is the latest version, there are no updates
+            if [ "$current_version" = "$latest_version" ]; then
+                next_version=""
+            fi
+
+             # If there is a next version (next version is not empty or is different from current version), save it to the targets.env file
+            if [ "$next_version" != "" ]; then
+                echo "${component_upper}_VERSION=$next_version" >> /home/system/components/updater/targets.env
```

Comment:
Should this path be releative?

Reply:
```suggestion
                echo "${component_upper}_VERSION=$next_version" >> ./targets.env
```

Just wondering: if we run the script from another folder (like `./path/script.sh`, would this work fine? Is the relative path in the script always relative to its script location?